source/GameInterface/Services/MobileParties/Patches/UnitRecruitPatch.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Common;
+using Common.Extensions;
+using Common.Logging;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Policies;
+using GameInterface.Services.MobileParties.Extensions;
+using GameInterface.Services.MobileParties.Handlers;
+using GameInterface.Services.MobileParties.Messages;
+using HarmonyLib;
+using Serilog;
+using Serilog.Core;
+using System;
+using System.Reflection;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Roster;
+using TaleWorlds.CampaignSystem.ViewModelCollection.CharacterDeveloper;
+
+namespace GameInterface.Services.MobileParties.Patches
+{
+    [HarmonyPatch(typeof(TroopRoster))]
+    public class UnitRecruitPatch
+    {
+        private static PropertyInfo TroopRoster_OwnerParty => typeof(TroopRoster).GetProperty("OwnerParty", BindingFlags.Instance | BindingFlags.NonPublic);
+
+        private static readonly MethodInfo TroopRoster_AddNewElement = typeof(TroopRoster).GetMethod("AddNewElement", BindingFlags.NonPublic | BindingFlags.Instance);
+
+        private static readonly ILogger Logger = LogManager.GetLogger<UnitRecruitPatch>();
+
+        [HarmonyPrefix]
+        [HarmonyPatch("AddNewElement")]
+        private static bool PrefixAddNewElement(TroopRoster __instance, CharacterObject character, bool insertAtFront = false, int insertionIndex = -1)
```

Comment:
I don't see how this patch on the client won't run with uncontrolled parties

Reply:
Almost sure this is the same as above